### PR TITLE
OSSL_PARAM_construct_utf8_string computes the string length.

### PR DESCRIPTION
**Pull request by paulidale**
If the passed string length is zero, the function computes the string length
from the passed string.

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: doc/man3/OSSL_PARAM_int.pod [hunk 1]
```diff
@@ -166,23 +166,22 @@ size B<rsize> is created.
 
 OSSL_PARAM_construct_utf8_string() is a function that constructs a UTF8
 string OSSL_PARAM structure.
-A parameter with name B<key>, storage B<buf>, size B<bsize> and return
-size B<rsize> is created.
+A parameter with name B<key>, storage B<buf> and size B<bsize> is created.
+If B<bsize> is zero, the string length is determined using strlen(3).
```
Comment (levitte) [inline doc/man3/OSSL_PARAM_int.pod]: A naked `strlen(3)` should prove interesting.  I just read in the [Pod::Man manual](https://perldoc.perl.org/Pod/Man.html) that it detects and formats these properly, so it should be absolutely fine, really.  It'll be interesting to see how Pod::Html treats it...

Comment (paulidale) [reply doc/man3/OSSL_PARAM_int.pod]: I'm willing to change it to whatever is required.

**Comment by paulidale** [comment]
I think I was overly aggressive about the changes, I've backed the `test/` ones out which ought to appease the CIs.  These didn't use `strlen(2)` to compute the length of the string, so the output has changed.

A re-approval is required.

**Comment by paulidale** [comment]
Merged, thanks.  This irksome little thing is gone now.

**Pull request closed without merging**